_posts/platform/internals/stacks/2000-01-01-stacks.md

```diff
@@ -0,0 +1,132 @@
+| [scalingo-14]({% post_url platform/internals/stacks/2000-01-01-scalingo-14-stack %}) | Ubuntu 14.04 | December 2019 | [scalingo/builder](https://hub.docker.com/r/scalingo/builder/) |
+
+All Scalingo buildpacks officially supported by Scalingo are compatible with
+all stacks.
+
+The default stack for all newly created applications is `scalingo-18`.
+
+Theese base Docker images are used for all applications hosted on the
+platform, as a result, it is a *generic image* which is *unspecialized*.
+That's why they are based on a stable **Ubuntu LTS** environment (currently
+18.04 LTS).
+
+## Viewing Stack Used by your Application
+
+The stack used by your application is available using Scalingo [CLI]({%
+post_url platform/cli/2000-01-01-start %}):
+
+```shell
+scalingo --app my-app apps-info
++----------------+-------------+
+|    SETTINGS    |    VALUE    |
++----------------+-------------+
+[...]
+| Stack          | scalingo-18 |
+[...]
++----------------+-------------+
+```
+
+## Migrating to a New Stack
+
+You can switch to the latest stack with:
+
+```shell
+scalingo --app my-app stacks-set scalingo-18
+```
+
+Then redeploy your application:
+
+```shell
+git commit --allow-empty -m "Upgrading to scalingo-18"
+git push scalingo master
+```
+
+You can view the list of all available stacks with the [CLI]({% post_url
+platform/cli/2000-01-01-start %}):
+
+```shell
+scalingo stacks
+```
+
+A nice solution to test the stack upgrade without affecting your currently 
+running application is to create a [review app]({% post_url
+platform/app/2000-01-01-review-apps %}) and add a `scalingo.json` file
+containing:
+
+```json
+{
+  "stack": "scalingo-18"
+}
+```
+
+Make sure your review app deploys successfully and works flawlessly. Once your
+application is ready for the new stack, switch to the desired stack and redeploy 
+with the method described above.
+
+## The build process
+
+Each time any user deploys a new release of their applications, a new Docker
+image is created. A new layer is added on top of the `scalingo/builder` base
+image. It's built using the buildpack and contains the application code and all
+its dependencies. When the build is done, the resulting Docker image is sent to
+our private repository and our orchestrator will use it subsequently to actually
+run the application in our infrastructure.
+
+The **maximum build time is 30 minutes**. After that duration the build will be canceled by the platform.
+
+## Installed libraries
+
+We've installed different libraries and softwares which are commonly used in
+the build process or used by human when running [one-off
+container]({% post_url platform/app/2000-01-01-tasks %}).
+
+You'll find:
+
+* Build essentials utility (GCC, make, autotools, ...)
+* Curl, git, telnet, ssh, ssh-client, openssl, dnsutils, sqlite
+* Node.js, Ruby, Perl, Python, Java
+* ImageMagick
+* MySQL, PostgreSQL MongoDB, Redis clients and development libraries
```

Comment:
```suggestion
```

This is not true anymore as we provide the customer with the script `dbclient-fetcher` to download DB clients in a one-off